### DISCUSSION TITLE
Relation between `maturin` and `cross-rs`?

### DISCUSSION THREAD

**Discussion by bluenote10** [PyO3/maturin#2287] [Q&A]
I'm new to the cross compiling world, and I'm frequently seeing recommendations to use [`cross`](https://github.com/cross-rs/cross).

I'm generally trying to understand what is the relationship between `maturin` and `cross`. Does `maturin` already use `cross` internally in one way or another? If not, would it make sense to somehow use them together, i.e., could `maturin` delegate compilation to `cross` for easier cross compilation?

Motivation: Whenever I'm encountering a situation that involves third party dependencies outside the manylinux scope (`libasound2`, `libdbus`, ...) I'm basically hitting a wall with cross compilation using `maturin`. So I was wondering if things would get easier with `cross` seeing e.g. [this example](https://github.com/cross-rs/cross/wiki/FAQ#linking-external-libraries).

**Comment by messense** [comment]
> Does `maturin` already use `cross` internally in one way or another?

No.

> If not, would it make sense to somehow use them together, i.e., could `maturin` delegate compilation to `cross` for easier cross compilation?

You can try set `CARGO=cross` but it's not well tested and may not work at all.
